src/transaction/confirm.rs

```diff
@@ -0,0 +1,601 @@
+                    }
+
+                    let starting_block_num = *unpinned.starting_block_num.get_or_insert(block_num);
+                    if block_num.saturating_sub(starting_block_num)
+                        > U256::from(unpinned.params.block_timeout)
+                    {
+                        return Poll::Ready(Err(ExecutionError::ConfirmTimeout));
+                    }
+
+                    ConfirmState::CreatingFilter(
+                        unpinned.web3.eth_filter().create_blocks_filter().compat(),
+                        target_block_num,
+                        remaining_confirmations.as_u64(),
+                    )
+                }
+                ConfirmState::CreatingFilter(ref mut create_filter, target_block_num, count) => {
+                    match ready!(Pin::new(create_filter).poll(cx)) {
+                        Ok(filter) => ConfirmState::WaitingForBlocks(
+                            filter
+                                .stream(unpinned.params.poll_interval)
+                                .skip(*count - 1)
+                                .into_future()
+                                .compat(),
+                        ),
+                        Err(_) => {
+                            // NOTE: In the case we fail to create a filter
+                            //   (usually because the node doesn't support pub/
+                            //   sub) then fall back to polling.
+                            ConfirmState::PollDelay(
+                                Delay::new(unpinned.params.poll_interval),
+                                *target_block_num,
+                            )
+                        }
+                    }
+                }
+                ConfirmState::WaitingForBlocks(ref mut wait) => {
+                    match ready!(Pin::new(wait).poll(cx)) {
+                        Ok(_) => ConfirmState::Check,
+                        Err((err, _)) => return Poll::Ready(Err(err.into())),
+                    }
+                }
+                ConfirmState::PollDelay(ref mut delay, target_block_num) => {
+                    ready!(Pin::new(delay).poll(cx));
+                    ConfirmState::PollCheckingBlockNumber(
+                        unpinned.web3.eth().block_number().compat(),
+                        *target_block_num,
+                    )
+                }
+                ConfirmState::PollCheckingBlockNumber(ref mut block_num, target_block_num) => {
+                    let block_num = match ready!(Pin::new(block_num).poll(cx)) {
+                        Ok(block_num) => block_num,
+                        Err(err) => return Poll::Ready(Err(err.into())),
+                    };
+
+                    if block_num == *target_block_num {
+                        ConfirmState::Checking(future::try_join(
+                            MaybeReady::ready(Ok(block_num)),
+                            unpinned
```

Comment:
This is where you were wondering if you can reuse the old tx hash once it's mined or if you have to always refetch it?